openedx/core/lib/block_structure/manager.py

```diff
@@ -90,39 +92,48 @@ def get_collected(self):
                 from each registered transformer.
         """
         try:
-            block_structure = BlockStructureFactory.create_from_cache(
+            block_structure = BlockStructureFactory.create_from_store(
                 self.root_block_usage_key,
-                self.block_structure_cache
+                self.store,
             )
             BlockStructureTransformers.verify_versions(block_structure)
 
         except (BlockStructureNotFound, TransformerDataIncompatible):
-            block_structure = self.update_collected()
+            if config.is_enabled(config.RAISE_ERROR_WHEN_NOT_FOUND):
```

Comment:
**Note to reviewers:** We now plan to raise an error instead, if the block-structure is not available when requested.  This is now an unexpected situation since the block-structure now has permanent storage.  We honor the `RAISE_ERROR_WHEN_NOT_FOUND ` waffle switch in the interim as we transition over to the new strategy.

Reply:
@nasthagiri sorry to spin up something from 4 years ago, has there been any change in plan, or is `RAISE_ERROR_WHEN_NOT_FOUND` still a temporary solution?